src/kakarot/accounts/eoa/library.cairo

```diff
@@ -140,8 +140,22 @@ namespace ExternallyOwnedAccount {
             assert [current_tx_calldata] = payload_len;
             assert offset = 1;
             assert selector = DEPLOY_CONTRACT_ACCOUNT;
+            tempvar syscall_ptr = syscall_ptr;
+            tempvar range_check_ptr = range_check_ptr;
             // Else run the bytecode of the destination contract
         } else {
+            // transfer the amount from the call to the destination
+            let (native_token_address_) = IKakarot.get_native_token(
+                contract_address=_kakarot_address
+            );
+            let (success) = IEth.transfer(
+                contract_address=native_token_address_,
+                recipient=destination,
+                amount=Uint256(amount, 0),
```

Comment:
⚠️ `amount` is a `felt` and there's absolutely no guarantee that `amount < 2**128` but your code will never break as it is possible to create a `Uint256` with a `low > 2**128`

Reply:
furthermore, you probably want to transfer only if `amount > 0` to save a useless tx